File: Algoritmos/1º Período/java/L3e3_NumPerf.java
/*
 * Autor: Matheus Scaketti
 * Data: 06/04/2016
 * 
 * Lista: 2
 * Exercício: 3
 * 
 * Descrição: Número perfeito é aquele cuja soma de seus divisores, 
 * exceto ele próprio, é igual a ele mesmo.
 * Ex: 6 é perfeito porque 1 + 2 + 3 = 6. Determine dentre 10(dez) 
 * números fornecidos pelo usuário quais são ou não perfeito.
 * 
*/

import java.util.Scanner;

public class L3e3_NumPerf {

	public static void main(String[] args) {
		
		int x, h, numSoma = 1;
		int[] nums = new int[10];
		Scanner l = new Scanner(System.in);
		
		System.out.println("Digite 10 números aaleatórios:");
		for (x = 0; x <= 9; x++){
			nums[x] = l.nextInt();
		}
		l.close();
		
		for (h = 0; h <= 9; h++){
			for(x = 2; x < nums[h]; x++){
				
				if (nums[h]%x == 0){
					numSoma += x;
				}
			}
			if(numSoma == nums[h]){
				System.out.println(nums[h] + " é perfeito.");
			}else{
				System.out.println(nums[h] + " não é perfeito.");
			}
			numSoma = 1;
		}
	}

}

File: Algoritmos/1º Período/C/L3e3_NumPerf.c
/*
 * Autor: Alexandre Mellado
 * Data: 16/04/2016
 * 
 * Lista: 3
 * Exercício: 3
 * 
 * Descrição: Número perfeito é aquele cuja soma de seus divisores, 
 * exceto ele próprio, é igual a ele mesmo.
 * Ex: 6 é perfeito porque 1 + 2 + 3 = 6. Determine dentre 10(dez) 
 * números fornecidos pelo usuário quais são ou não perfeito.
 * 
*/

#include <stdio.h>
#include <stdlib.h>

int main(){
	int n,i,x,aux;
	for ( aux = 1 ; aux <= 10 ; aux++ ){
		x = 0;
		printf("Digite um numero: ");
		scanf("%d",&n);
		for ( i = (n - 1) ; i>0 ; i-- )
			if (n%i == 0)
				x = x + i;
		if (x == n)
			printf("%d eh um numero perfeito.\n",n);
		else
			printf("%d nao eh um numero perfeito.\n",n);
			printf("  ---------------- \n");
	}
	getch();
	return 0;  
}
